src/pq_flash_index.cpp

```diff
@@ -65,7 +65,7 @@ namespace diskann {
 
   template<typename T>
   QueryScratch<T>::QueryScratch(size_t aligned_dim, size_t visited_reserve) {
-    _u64 coord_alloc_size = ROUND_UP(MAX_N_CMPS * aligned_dim, 256);
+    _u64 coord_alloc_size = ROUND_UP(MAX_N_CMPS * aligned_dim * sizeof(T), 256);
```

Comment:
In the DLVS code, we removed the MAX_N_CMPS from the expression to save space. IIRC, coord_scratch only holds one point at a time. Maybe you want to do the same here? 
In the DLVS code, it is simply aligned_dim * sizeof(T).

Reply:
That is a Bug, If buf_idx > 4069 will coredump

Reply:
@gopalrs Let's take the reduction of the MAX_N_CMPS as a separate issue.